### ISSUE TITLE
Unable to output single line array syntax

### ISSUE THREAD

**Issue by mabushey** [dtolnay/serde-yaml#355]
❤️ First off, serde is amazing. :)

I'm using `serde_yaml::to_writer()` to dump a struct to a yaml file. In the struct I have `tmdb_genre: Vec<u64>` which outputs like this:
```
tmdb_genre:
- 35
- 10751
- 9648
- 10749
```
While this is correct yaml, for purely aesthetic reasons I would like it to look like this: 
```
tmdb_genre: [35, 10751, 9648, 10749]
```

I tried adding `#[serde(flatten)]` to the struct element but I get  runtime Error("can only flatten structs and maps (got a sequence)"). Is there any way of doing this w/o resorting to post processing of the yaml (ie sed)? Thanks!

**Comment by AurelienFT** [comment]
Did you found the solution?

**Comment by mabushey** [comment]
> Did you found the solution?

Nope, doing a system call to sed afterwards. Thanks for asking. :)
